### PR TITLE
MINIFICPP-1391 Upgrade XCode version and replace usages of set-env

### PR DESCRIPTION
**Pull request by fgerlits**
GitHub actions fixes:

- Remove the XCode 10.3 job, and add a new job with XCode 12.0
- Replace deprecated usages of `set-env` (see https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/)
- Remove the CMAKE_BUILD_OPTIONS env setting, as it doesn't work -- our CI jobs look like they set ENABLE_PCAP=TRUE, but they do not
- Increase the timeout on the Docker test, as it sometimes takes more than 60 minutes.

https://issues.apache.org/jira/browse/MINIFICPP-1391

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by lordgamez — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: "MiNiFi-CPP CI"
 on: [push, pull_request, workflow_dispatch]
 jobs:
-  macos_xcode_10_3:
-    name: "macos-xcode10.3"
+  macos_xcode_11_2_1:
+    name: "macos-xcode11.2.1"
```
Comment (arpadboda) [inline .github/workflows/ci.yml:5]: I'm not sure there is too much value in running two builds using the same major version. 

I would recommend one to utilize xcode 12:
https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#xcode

Comment (fgerlits) [reply .github/workflows/ci.yml:5]: I have replaced 11.7 with 12.0

**Review by hunyadi-dev — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,12 +13,11 @@ jobs:
...
-          echo "::set-env name=CMAKE_BUILD_OPTIONS::-DENABLE_PCAP=TRUE -DENABLE_LUA_SCRIPTING=1"
-          sudo xcode-select -switch /Applications/Xcode_10.3.app
+          sudo xcode-select -switch /Applications/Xcode_11.2.1.app
       - id: build
-        run: ./bootstrap.sh -e -t && cd build  && cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_RULE_MESSAGES=OFF -DSTRICT_GSL_CHECKS=AUDIT .. && cmake --build . --parallel 4 && make test ARGS="--timeout 300 -j4 --output-on-failure" && make linter
-  macos_xcode_11_2_1:
-    name: "macos-xcode11.2.1"
+        run: ./bootstrap.sh -e -t && cd build  && cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_LUA_SCRIPTING=1 -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_RULE_MESSAGES=OFF -DSTRICT_GSL_CHECKS=AUDIT .. && cmake --build . --parallel 4 && make test ARGS="--timeout 300 -j4 --output-on-failure" && make linter
+  macos_xcode_11_7:
+    name: "macos-xcode12.0"
```
Comment (arpadboda) [inline .github/workflows/ci.yml:20]: The section and the name seems to conflict

Comment (fgerlits) [reply .github/workflows/ci.yml:20]: Oops, good catch.  Fixed.

**Review by arpadboda — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -77,9 +75,8 @@ jobs:
             ubuntu-16.04-ccache-refs/heads/main-
       - id: install_deps
         run: |
-          echo "::set-env name=CMAKE_BUILD_OPTIONS::-DENABLE_PCAP=TRUE"
```
Comment (szaszm) [inline .github/workflows/ci.yml:80]: Is this unused?

**Review by szaszm — APPROVED** [review]

**Comment by fgerlits** [comment]
> Is this unused?

Cmake doesn't look at the value of `CMAKE_BUILD_OPTIONS`; possibly the cmake command line contained `... ${CMAKE_BUILD_OPTIONS} ...` at some point, but not recently.

So this line made it appear as if we were compiling with `ENABLE_PCAP=TRUE`, but we weren't really.

**Comment by szaszm** [comment]
I've deleted this question because I realized that you've already answered it in the description. Thanks for the clarification.

**Pull request closed without merging**